src/test/java/com/github/checkstyle/patchfilter/SuppressionPatchFilterTest.java

```diff
@@ -51,33 +58,166 @@ public void testAccept() throws Exception {
+                + "puppycrawl/tools/checkstyle/grammar/"
+                + "java14/InputJava14InstanceofWithPatternMatching.java "
+                + "b/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/grammar/"
+                + "java14/InputJava14InstanceofWithPatternMatching.java", file2.getDiffCommand());
+        assertEquals("InputJava14InstanceofWithPatternMatching.java", file2.getFromFile());
+        assertEquals("InputJava14InstanceofWithPatternMatching.java", file2.getToFile());
+        assertEquals(1, file2.getPatch().getDeltas().size());
+        assertNull(diff.getTail());
+    }
+
+    @Test
+    public void testGitFormatPatchLinuxThree() throws Exception {
+        final String fileName = getPath("PatchFileFromDiffTools/GitFormatPatchLinuxThree.txt");
+        final FileInputStream ins = new FileInputStream(fileName);
+        final UnifiedDiff diff = UnifiedDiffReader.parseUnifiedDiff(ins);
+        assertEquals(1, diff.getFiles().size());
+        final UnifiedDiffFile file1 = diff.getFiles().get(0);
+        assertEquals("diff --git a/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/"
+                + "JavadocMethodCheck.java"
+                + " b/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/"
+                + "JavadocMethodCheck.java", file1.getDiffCommand());
+        assertEquals("src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/"
+                + "JavadocMethodCheck.java", file1.getFromFile());
+        assertEquals("src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/"
+                + "JavadocMethodCheck.java", file1.getToFile());
+        assertEquals(1, file1.getPatch().getDeltas().size());
+        assertNull(diff.getTail());
+    }
+
+    @Test
+    public void testGitFormatPatchWindows() throws Exception {
+        final String fileName = getPath("PatchFileFromDiffTools/GitFormatPatchWindows.txt");
+        final FileInputStream ins = new FileInputStream(fileName);
+        final UnifiedDiff diff = UnifiedDiffReader.parseUnifiedDiff(ins);
+        assertEquals(1, diff.getFiles().size());
+        final UnifiedDiffFile file1 = diff.getFiles().get(0);
+        assertEquals("diff --git a/src/main/java/com/puppycrawl/tools/checkstyle/Checker.java "
+                        + "b/src/main/java/com/puppycrawl/tools/checkstyle/Checker.java",
+                file1.getDiffCommand());
+        assertEquals("src/main/java/com/puppycrawl/tools/checkstyle/Checker.java",
+                file1.getFromFile());
+        assertEquals("src/main/java/com/puppycrawl/tools/checkstyle/Checker.java",
+                file1.getToFile());
+        assertEquals(1, file1.getPatch().getDeltas().size());
+        assertEquals("2.25.1.windows.1\n\n", diff.getTail());
+    }
+
+    @Test
+    @Ignore("filename parsed error, Expected :<tests/test-check-pyflakes.t>, "
+            + "Actual :<tests/test-check-pyflakes.t\tTue Jun 09 17:13:26 2020 -0400>\n")
```

Comment:
same as above, please put a link to a lib issue

Reply:
done, I make a issue to java-diff-utils add "until https://github.com/java-diff-utils/java-diff-utils/issues/85"